src/api/TradeAPI.ts

```diff
@@ -113,4 +134,130 @@ export default class TradeAPI {
+   *
+   * @return {Promise<TradeQuote>}
+   */
+  public async fetchTradeQuoteAsync(
+    fromToken: Address,
+    toToken: Address,
+    fromTokenDecimals: number,
+    toTokenDecimals: number,
+    rawAmount: string,
+    fromAddress: Address,
+    setToken: SetTokenAPI,
+    gasPrice?: number,
+    slippagePercentage?: number,
+    isFirmQuote?: boolean,
+    feePercentage?: number,
+    feeRecipient?: Address,
+    excludedSources?: string[],
+  ): Promise<TradeQuote> {
+    this.assert.schema.isValidAddress('fromToken', fromToken);
+    this.assert.schema.isValidAddress('toToken', toToken);
+    this.assert.schema.isValidAddress('fromAddress', fromAddress);
+    this.assert.schema.isValidJsNumber('fromTokenDecimals', fromTokenDecimals);
+    this.assert.schema.isValidJsNumber('toTokenDecimals', toTokenDecimals);
+    this.assert.schema.isValidString('rawAmount', rawAmount);
+
+    const chainId = (await this.provider.getNetwork()).chainId;
+
+    return this.tradeQuoter.generate({
+      fromToken,
+      toToken,
+      fromTokenDecimals,
+      toTokenDecimals,
+      rawAmount,
+      fromAddress,
+      chainId,
+      tradeModule: this.tradeModuleWrapper,
+      provider: this.provider,
+      setToken,
+      gasPrice,
+      slippagePercentage,
+      isFirmQuote,
+      feePercentage,
+      feeRecipient,
+      excludedSources,
+    });
+  }
+
+  /**
+   * Fetches a list of tokens and their metadata from CoinGecko. Each entry includes
+   * the token's address, proper name, decimals, exchange symbol and a logo URI if available.
+   * For Ethereum, this is a list of tokens tradeable on Uniswap, for Polygon it's a list of
+   * tokens tradeable on Sushiswap's Polygon exchange. Method is useful for acquiring token decimals
+   * necessary to generate a trade quote and images for representing available tokens in a UI.
+   *
+   * @return List of tradeable tokens for chain platform
+   */
+  public async fetchTokenListAsync(): Promise<CoinGeckoTokenData[]> {
+    await this.initializeForChain();
+    return this.coinGecko.fetchTokenList();
+  }
+
+  /**
+   * Fetches the same info as `fetchTokenList` in the form of a map indexed by address. Method is
+   * useful if you're cacheing the token list and want quick lookups for a variety of trades.
+   *
+   * @return Map of token addresses to token metadata
+   */
+  public async fetchTokenMapAsync(): Promise<CoinGeckoTokenMap> {
+    await this.initializeForChain();
+    return this.coinGecko.fetchTokenMap();
+  }
+
+  /**
+   * Fetches a list of prices vs currencies for the specified inputs from CoinGecko
+   *
+   * @param  contractAddresses         String array of contract addresses
+   * @param  vsCurrencies              String array of currency codes (see CoinGecko api for a complete list)
+   *
+   * @return                           List of prices vs currencies
+   */
+  public async fetchCoinPricesAsync(
```

Comment:
I think most of these below belong in a different API, perhaps ERC20 API or something of the sort